### PR TITLE
add transformer for emit add property to default export

### LINKED ISSUE #29206: Attaching property to exported function creates invalid definition file
**TypeScript Version:**  3.3.0-dev.20181229

**Search Terms:**
default export function property namespace

**Code**

```ts
export default function someFunc() {
    return 'hello!';
}

someFunc.someProp = 'yo';
```

**Expected behavior:**
The declaration file should be valid

**Actual behavior:**
```
packages/proj-1/build/index.d.ts:2:26 - error TS1005: ';' expected.

2 export default namespace someFunc {
                           ~~~~~~~~

packages/proj-1/build/index.d.ts:2:35 - error TS1005: ';' expected.

2 export default namespace someFunc {
```

**Playground Link:** Cannot be reproduced in the playground as the issue is in the `d.ts` file, not the JS. However, I've setup a repository: https://github.com/SimenB/tsc-b-auch

**Related Issues:** None found

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #29206

### CODE REVIEW

File: src/compiler/transformers/declarations.ts [hunk 1]
```diff
@@ -1008,7 +1008,43 @@ namespace ts {
...
+                            clean.type,
+                            /*body*/ undefined
+                        );
+
+                        const namespaceDeclaration = updateModuleDeclaration(
+                            namespaceDecl,
+                            /*decorators*/ undefined,
+                            createModifiersFromModifierFlags(modifierFlags),
+                            input.name!,
+                            createModuleBlock(declarations)
```
Comment (weswigham) [inline src/compiler/transformers/declarations.ts]: `namespaceDecl.body` should save some work here, no?

**Comment by DanielRosenwasser** [comment]
```
ERROR: /home/travis/build/Microsoft/TypeScript/src/compiler/transformers/declarations.ts:1039:42 boolean-trivia There should be 1 space between an argument and its comment.
ERROR: /home/travis/build/Microsoft/TypeScript/src/compiler/transformers/declarations.ts:1040:47 boolean-trivia There should be 1 space between an argument and its comment.
```

**Comment by Kingwl** [comment]
⬆️

File: src/compiler/types.ts [hunk 2]
```diff
@@ -3508,7 +3508,8 @@ namespace ts {
         | ModuleDeclaration
         | TypeAliasDeclaration
         | InterfaceDeclaration
-        | EnumDeclaration;
+        | EnumDeclaration
+        | ExportAssignment;
```
Comment (weswigham) [inline src/compiler/types.ts]: Why was this added here (and the empty case in `transformTopLevelDeclaration`)? Without it being added to the `isLateVisibilityPaintedStatement`, it will still fail to be printed late.

**Comment by Kingwl** [comment]
🆙
